aptos-move/move-examples/data_structures/sources/smart_table.move

```diff
@@ -0,0 +1,358 @@
+/// A smart table implementation based on linear hashing. (https://en.wikipedia.org/wiki/Linear_hashing)
+/// Compare to Table, it uses less storage slots but has higher chance of collision, it's a trade-off between space and time.
+/// Compare to other implementation, linear hashing splits one bucket a time instead of doubling buckets when expanding to avoid unexpected gas cost.
+/// SmartTable uses faster hash function SipHash instead of cryptographically secure hash functions like sha3-256 since it tolerates collisions.
+module aptos_std::smart_table {
+    use std::error;
+    use std::vector;
+    use aptos_std::aptos_hash::sip_hash_from_value;
```

Comment:
Why sip hash specifically? What are the considerations here (collision, gas cost, etc.)?

Reply:
good perf and prevention of hash flooding attack.